src/rgb/contract/conceal.rs

```diff
@@ -22,21 +23,77 @@ pub trait AutoConceal {
     fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize;
 }
 
-// TODO: Do an auto implementation for and other collection types
-
 impl<T> AutoConceal for Vec<T>
 where
     T: AutoConceal + Ord + Clone,
 {
     fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize {
         let mut count = 0;
-        let mut new_self = Vec::<T>::new();
+        for item in self.iter_mut() {
+            count += item.conceal_except(seals);
+        }
+        count
+    }
+}
+
+impl<T> AutoConceal for BTreeSet<T>
+where
+    T: AutoConceal + Ord + Clone,
+{
+    fn conceal_except(&mut self, seals: &Vec<seal::Confidential>) -> usize {
+        let mut count = 0;
+        let mut new_self = BTreeSet::<T>::new();
```

Comment:
Not sure whether `BTreeSet` (or `HashSet`) provides `with_capacity()`, but if it does this is important to use it, otherwise we may get a re-allocation per each function cycle.

Reply:
It seems both the sets don't provide `with_capacity()` so we are stuck with this. :(